qemu.go

```diff
@@ -1018,31 +1018,25 @@ func (q *qemu) resumePod() error {
 
 // addDevice will add extra devices to Qemu command line.
 func (q *qemu) addDevice(devInfo interface{}, devType deviceType) error {
-	switch devType {
-	case fsDev:
-		volume := devInfo.(Volume)
-		q.qemuConfig.Devices = q.appendVolume(q.qemuConfig.Devices, volume)
-	case serialPortDev:
-		socket := devInfo.(Socket)
-		q.qemuConfig.Devices = q.appendSocket(q.qemuConfig.Devices, socket)
-	case netDev:
-		endpoint := devInfo.(Endpoint)
-		q.qemuConfig.Devices = q.appendNetwork(q.qemuConfig.Devices, endpoint)
-	case blockDev:
-		drive := devInfo.(Drive)
-		q.qemuConfig.Devices = q.appendBlockDevice(q.qemuConfig.Devices, drive)
-	case vhostuserDev:
-		switch vhostUserDev := devInfo.(type) {
-		case VhostUserNetDevice:
-			q.qemuConfig.Devices = q.appendVhostUserDevice(q.qemuConfig.Devices, &vhostUserDev)
-		case VhostUserSCSIDevice:
-			q.qemuConfig.Devices = q.appendVhostUserDevice(q.qemuConfig.Devices, &vhostUserDev)
-		case VhostUserBlkDevice:
-			q.qemuConfig.Devices = q.appendVhostUserDevice(q.qemuConfig.Devices, &vhostUserDev)
-		}
-	case vfioDev:
-		vfDevice := devInfo.(VFIODevice)
-		q.qemuConfig.Devices = q.appendVFIODevice(q.qemuConfig.Devices, vfDevice)
+	switch v := devInfo.(type) {
+	case Volume:
+		q.qemuConfig.Devices = q.appendVolume(q.qemuConfig.Devices, v)
+	case Socket:
+		q.qemuConfig.Devices = q.appendSocket(q.qemuConfig.Devices, v)
+	case Endpoint:
+		q.qemuConfig.Devices = q.appendNetwork(q.qemuConfig.Devices, v)
+	case Drive:
+		q.qemuConfig.Devices = q.appendBlockDevice(q.qemuConfig.Devices, v)
+
+	//vhostUserDevice is an interface, hence the pointer for Net, SCSI and Blk:
```

Comment:
Could not you simplify like this:
```go
        case VhostUserNetDevice, VhostUserSCSIDevice, VhostUserBlkDevice:
		q.qemuConfig.Devices = q.appendVhostUserDevice(q.qemuConfig.Devices, &v)
```

Reply:
I like the refactoring BTW.

Reply:
No, you cannot in the case of a type switch, unfortunately.  It pained me to not be able to!

Reply:
You can use "fallthrough" in a switch iirc.

Reply:
The issue is that the resolution of &v isn't carried out correctly when it isn't explicitly done this way (because it is a type switch and VhostUserDevice is an interface).

Reply:
Interesting, didnt know the "fallthrough" keyword didnt work for type switches.